packages/datadog-plugin-jest/test/index.spec.js

```diff
@@ -15,19 +20,20 @@ describe('Plugin', () => {
       tracer = require('../../dd-trace')
       return agent.load('jest').then(() => {
         DatadogJestEnvironment = require(`../../../versions/jest-environment-node@${version}`).get()
+        datadogJestEnv = new DatadogJestEnvironment({ rootDir: BUILD_SOURCE_ROOT }, { testPath: TEST_SUITE })
+        datadogJestEnv.context.expect = {
+          getState: () => {
+            return {
+              currentTestName: TEST_NAME
+            }
+          }
+        }
```

Comment:
`expect` is injected a posteriori on `jest` so we need to set it up ourselves in the test

Reply:
Any way to test that this works without a mock? This test does not guarantee that this actually works the way it should.

Reply:
this relates to what we talked about when merging the `jest` instrumentation. Ideally we would instrument the runner, the one that instantiates the environment. This way we could test that. As of now, that is not possible.

Reply:
in case you're curious. The `expect` we use to get the current test name is injected in https://github.com/facebook/jest/blob/master/packages/jest-circus/src/legacy-code-todo-rewrite/jestAdapterInit.ts#L132, which is called in https://github.com/facebook/jest/blob/master/packages/jest-runner/src/runTest.ts#L251 (within the `jest-runner` code)

Reply:
I'm definitely not a fan that they make it very clear that they will rewrite that and we're using a mock so we won't know when it happens, but for now I guess we don't have a choice as you said. Can you add a TODO to make sure we don't forget to rewrite the test when the runner can be instrumented directly?

Reply:
yeah I agree it's far from ideal. I will add the TODO 👍